src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1465,6 +1478,37 @@ protected void ExecuteIndexing()
             }
         }
 
+        private void FinishStaggeredDeploymentForThisNode(IndexDefinitionBase definition)
+        {
+            var nodeTag = DocumentDatabase.ServerStore.NodeTag;
+            var record = DocumentDatabase.ReadDatabaseRecord();
+
+
+            if (record.StaggeredIndexes.TryGetValue(definition.Name, out var staggeredIndex) == false)
+                return;
+
+            if (staggeredIndex.TryGetValue(nodeTag, out var staggeredIndexStatus) == false)
+                return;
+
+            if (staggeredIndexStatus != StaggeredIndexStatus.Running)
+                return;
+
+            // todo We need to make sure we only run this command once for each finished node
```

Comment:
Why does that matter? what would happen if this runs multiple times?

Reply:
You generally can't tell if you have a leader or not.
But we don't need a `Task.Run`, do we? We can just run the `SendToLeaderAsync` and not wait on the task?
But we need to handle two error locations then.

Reply:
Can you please elaborate on "two error locations"?

Reply:
One for sync failure and one for async failure.